### DISCUSSION TITLE
[Script request] Traccar update script

### DISCUSSION THREAD

**Discussion by 391a** [community-scripts/ProxmoxVE#332] [Request script]
### Application Name

Traccar Modern GPS Tracking Platform

### Website

https://www.traccar.org/

### Description

Hello. Can you add an update to the existing traccar script? Thank you.
https://www.traccar.org/upgrading-traccar/
https://www.traccar.org/linux/
https://www.traccar.org/install-digitalocean/

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by 391a** [comment]
#!/usr/bin/env bash

# Copyright (c) 2021-2024 tteck
# Author: tteck (tteckster)
# License: MIT
# https://github.com/community-scripts/ProxmoxVE/raw/main/LICENSE

source /dev/stdin <<< "$FUNCTIONS_FILE_PATH"
color
verb_ip6
catch_errors
setting_up_container
network_check
update_os

msg_info "Installing/Updating Traccar"

# Check if curl and unzip are installed. Install them if not.
if ! command -v curl &>/dev/null || ! command -v unzip &>/dev/null; then
  msg_info "Installing Dependencies"
  $STD apt-get update # Update package lists
  $STD apt-get install -y curl unzip # Install curl and unzip
  msg_ok "Installed Dependencies"
fi

# Get the currently installed version (if Traccar is installed)
if [[ -f /opt/traccar/VERSION ]]; then
  INSTALLED_VERSION=$(cat /opt/traccar/VERSION)
else
  INSTALLED_VERSION=""
fi

# Get the latest available version from GitHub API
RELEASE=$(curl -s https://api.github.com/repos/traccar/traccar/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')

msg_info "Latest Traccar Version: ${RELEASE}"
if [[ ! -z "$INSTALLED_VERSION" ]]; then
  msg_info "Installed Traccar Version: ${INSTALLED_VERSION}"
fi

# Compare versions
if [[ "$RELEASE" == "$INSTALLED_VERSION" ]]; then
  msg_info "Traccar is already up to date."
  exit 0 # Exit the script if already up to date
fi

# Download and install/update Traccar
msg_info "Downloading Traccar v${RELEASE}"
wget -q https://github.com/traccar/traccar/releases/download/v${RELEASE}/traccar-linux-64-${RELEASE}.zip

msg_info "Installing/Upgrading Traccar"
if [[ -d /opt/traccar ]]; then
  # Upgrade: Stop the service before upgrading
    msg_info "Stopping Traccar Service"
    systemctl stop traccar
    msg_ok "Stopped Traccar Service"

  # Create a backup of the current installation (optional but recommended)
  msg_info "Backing up current Traccar installation"
  mv /opt/traccar /opt/traccar.bak
  msg_ok "Backup created at /opt/traccar.bak"
fi

# Extract the archive to /opt/ and rename the extracted directory
$STD unzip -o traccar-linux-64-${RELEASE}.zip -d /opt/
mv /opt/traccar-linux-64-${RELEASE} /opt/traccar

echo "${RELEASE}" > /opt/traccar/VERSION # Save the installed version

# Start the service. Check if service file exists first.
if [[ -f /etc/systemd/system/traccar.service ]]; then
    msg_info "Starting Traccar Service"
    systemctl enable --now traccar
    msg_ok "Started Traccar Service"
else
    msg_info "Creating systemd service"
    cat <<EOF >/etc/systemd/system/traccar.service
[Unit]
Description=Traccar GPS Tracking Server
After=network.target

[Service]
Type=simple
User=root # Change to the user Traccar should run as if necessary
WorkingDirectory=/opt/traccar
ExecStart=/opt/traccar/bin/start.sh
Restart=on-failure

[Install]
WantedBy=multi-user.target
EOF
    systemctl enable --now traccar
    msg_ok "Created and started systemd service"
fi

rm -rf README.txt traccar-linux-64-${RELEASE}.zip # No need to remove traccar.run, it is inside the archive

# Cleanup
msg_info "Cleaning up"
if [[ -d /opt/traccar.bak ]]; then
    rm -rf /opt/traccar.bak # Remove the old backup
    msg_ok "Removed old backup"
fi
$STD apt-get -y autoremove
$STD apt-get -y autoclean
msg_ok "Cleaned"

msg_ok "Traccar successfully installed/updated to v${RELEASE}"
motd_ssh
customize

**Comment by lipunis** [comment]
✔️  Core dependencies installed
curl: (22) The requested URL returned error: 403

[ERROR] in line 16: exit code 0: while executing command RELEASE=$(curl -fsSL https://api.github.com/repos/traccar/traccar/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
curl: (22) The requested URL returned error: 400
/dev/fd/63: line 72: SPINNER_PID: unbound variable

**Reply by michelroegl-brunner** [reply]
Open a Bug report issue pls and fill out the form

**Reply by tremor021** [reply]
![image](https://github.com/user-attachments/assets/9bc957d9-3dbe-4dc9-8f3f-2edd9cd83d47)
![image](https://github.com/user-attachments/assets/355a5d08-f01a-4e5a-9972-33715ea45751)
works fine

**Reply by MickLesk** [reply]
Curl 22 - 400
Bad Request, Network issue

**Comment by lipunis** [comment]
I am install two other apps after. I'll try again tomorrow

чт, 8 мая 2025 г., 17:05 CanbiZ ***@***.***>:
